src/mito2/src/worker/handle_alter.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Handling alter related requests.
+
+use std::sync::Arc;
+
+use common_query::Output;
+use common_telemetry::{error, info};
+use snafu::ResultExt;
+use store_api::metadata::{RegionMetadata, RegionMetadataBuilder, RegionMetadataRef};
+use store_api::region_request::RegionAlterRequest;
+use store_api::storage::RegionId;
+use tokio::sync::oneshot;
+
+use crate::error::{InvalidMetadataSnafu, InvalidRegionRequestSnafu, RegionNotFoundSnafu, Result};
+use crate::flush::FlushReason;
+use crate::manifest::action::{RegionChange, RegionMetaAction, RegionMetaActionList};
+use crate::memtable::MemtableBuilderRef;
+use crate::region::version::Version;
+use crate::region::MitoRegionRef;
+use crate::request::{DdlRequest, SenderDdlRequest};
+use crate::worker::{send_result, RegionWorkerLoop};
+
+impl<S> RegionWorkerLoop<S> {
+    pub(crate) async fn handle_alter_request(
+        &mut self,
+        region_id: RegionId,
+        request: RegionAlterRequest,
+        sender: Option<oneshot::Sender<Result<Output>>>,
+    ) {
+        let Some(region) = self.regions.get_region(region_id) else {
+            send_result(sender, RegionNotFoundSnafu { region_id }.fail());
+            return;
+        };
```

Comment:
I found that this code is duplicated at many places in worker implementation.

Maybe we can add a macro to simplify it, such as 
```rust
let Some(region) = try_get_region!(region_id, sender);
```
I am not sure if it can work in this way.

Reply:
A macro needs to contain the entire `let-else` expression. Some places don't send results to the sender directly so we can't reuse the same `try_get_region()` macro.

Reply:
Yes, but i found several places that have the same logic.
<img width="777" alt="Screenshot 2023-09-11 at 20 06 24" src="https://github.com/GreptimeTeam/greptimedb/assets/14142/6106ec8a-6827-4bb6-88b4-fda015ceda83">


Just a suggestion, feel free to accept it or not.